src/basic/fs-util.c

```diff
@@ -799,7 +799,8 @@ int chase_symlinks(const char *path, const char *original_root, unsigned flags,
                         return -ENOMEM;
         }
 
-        *ret = done;
+        if (ret)
+                *ret = done;
         done = NULL;
```

Comment:
There is a memleak:
```

FAIL: test-stat-util
====================


=================================================================
==28352==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 39 byte(s) in 1 object(s) allocated from:
    #0 0x7f81c53f6200 in realloc (/lib64/libasan.so.3+0xc7200)
    #1 0x7f81c4e85895 in strextend src/basic/string-util.c:726
    #2 0x7f81c4e990fa in chase_symlinks src/basic/fs-util.c:785
    #3 0x7f81c4e9b4f8 in path_is_os_tree src/basic/stat-util.c:158
    #4 0x561938aa9b04 in test_path_is_os_tree src/test/test-stat-util.c:64
    #5 0x561938aa9bd1 in main src/test/test-stat-util.c:72
    #6 0x7f81c47a7400 in __libc_start_main (/lib64/libc.so.6+0x20400)

SUMMARY: AddressSanitizer: 39 byte(s) leaked in 1 allocation(s).
FAIL test-stat-util (exit status: 1)
```
Should be
```c
if (ret) {
        *ret = done;
        done = NULL;
}
```
BTW: I have no idea how to review such giant PRs :-)